bentoml/_internal/server/http_app.py

```diff
@@ -264,6 +264,15 @@ def on_startup(self) -> list[t.Callable[[], None]]:
         else:
             for runner in self.bento_service.runners:
                 on_startup.append(runner.init_client)
+
+        async def wait_for_runner_ready():
+            ready_status = False
+            while not ready_status:
+                ready_status = all(await asyncio.gather(*(runner.runner_handle_is_ready() for runner in self.bento_service.runners)))
+                await asyncio.sleep(0.1)
+        
+        on_startup.append(lambda: asyncio.run(wait_for_runner_ready()))
```

Comment:
This won't work since you are calling `asyncio.run` inside a running event loop.
```suggestion
        async def wait_runner_ready() -> None:
            ready_status = False
            while not ready_status:
                ready_status = all(
                    await asyncio.gather(
                        *(
                            runner.runner_handle_is_ready()
                            for runner in self.bento_service.runners
                        )
                    )
                )

        on_startup.append(
            lambda: asyncio.ensure_future(
                wait_runner_ready(), loop=asyncio.get_running_loop()
            )
        )
```